dlt/_workspace/helpers/runtime/runtime_artifacts.py

```diff
@@ -0,0 +1,191 @@
+"""Implements SupportsTracking"""
+from typing import Any, ClassVar, List, Optional, Tuple, Union
+import fsspec
+import pickle
+import os
+
+import dlt
+from dlt.common import logger
+from dlt.common.configuration.exceptions import ConfigurationException
+from dlt.common.configuration.resolve import resolve_configuration
+from dlt.common.configuration.specs.base_configuration import BaseConfiguration, configspec
+from dlt.common.storages.configuration import FilesystemConfiguration
+from dlt.common.storages.fsspec_filesystem import FileItemDict, fsspec_from_config, glob_files
+from dlt.common.versioned_state import json_encode_state
+
+from dlt.pipeline.trace import PipelineTrace, PipelineStepTrace, TPipelineStep, SupportsPipeline
+from dlt._workspace.run_context import DEFAULT_WORKSPACE_WORKING_FOLDER
+from dlt._workspace._workspace_context import WorkspaceRunContext
+
+
+@configspec
+class RuntimeArtifactsConfiguration(BaseConfiguration):
+    artifacts: FilesystemConfiguration = None
+
+
+def sync_from_runtime() -> None:
```

Comment:
Two notes here:
* In the old implementation, we sometimes got some kind of "e-tag expired" exception. I think what happens is, that you do a ls of existing blobs and e-tags are created, but if in the meantime a pipeline uploads new stuff, they expire and the e-tags invalidate, although the blob-path still exists. Afaik there is a setting to supress this warning.
* This sync happens everytime you open the dashboard. Ideally we only sync files that have changed since the last sync, often there will be no changes at all if a user switches tabs a lot in the runtime web-ui. I did not have time to implement anything like this and it is not top prio, but we should have it at some point.